File: leet_code/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dataset={}
        for index in range(len(nums)):
            if (target - nums[index]) in dataset:
                return [index, dataset[(target-nums[index])]]
            else:
                dataset[(nums[index])] = index

File: leet_code/cpp/1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

#include <iostream>
#include <vector>
#include <map>
#include <algorithm>
using namespace std;

// @lc code=start
class Solution 
{
public:
    vector<int> twoSum(vector<int>& nums, int target) 
    {
        map<int, int> record;
        vector<int> res;
        for(auto i = 0; i < nums.size(); i++)
        {
            if(record.find(target-nums[i]) != record.end())
            {
                return {i, record[target-nums[i]]};
            }
            record[nums[i]] = i;
        }
        return res;
    }
};

// @lc code=end

